stdlib/public/Concurrency/TaskStatus.cpp

```diff
@@ -610,76 +768,26 @@ static swift_task_escalateImpl(AsyncTask *task, JobPriority newPriority) {
     // TODO (rokhinip): Add a signpost to flag that this is a potential
     // priority inversion
     SWIFT_TASK_DEBUG_LOG("[Override] Escalating %p which is enqueued", task);
+
   } else {
     SWIFT_TASK_DEBUG_LOG("[Override] Escalating %p which is suspended to %#x", task, newPriority);
+    // We must have at least one record - the task dependency one.
+    assert(newStatus.getInnermostRecord() != NULL);
   }
-#endif
 
   if (newStatus.getInnermostRecord() == NULL) {
     return newStatus.getStoredPriority();
   }
 
-  withStatusRecordLock(task, LockContext::OnTask, newStatus, [&] {
+  withStatusRecordLock(task, LockContext::OnTask, newStatus, [&](ActiveTaskStatus status) {
     // Perform escalation operations for all the status records.
-    for (auto cur: newStatus.records()) {
+    for (auto cur: status.records()) {
```

Comment:
Is this safe against reentrant changes to the status links, or is that not possible?

Reply:
Today, only the following actions can modify the status record list by adding or removing records:
* [add] Creating a child task - async let or task group
* [remove] Having the child task/task group complete
* [add] Suspending the task
* [remove] Making it runnable again
* [add] Installing a cancellation handler
* [remove] Running a cancellation handler

performEscalationAction doesn't really do any of the above and so it's not possible.

Reply:
Note that performCancellationAction however, does open us up to this since it calls out to client code in the cancellation handler which could do any of the above.